simulator/ns-3.35/src/point-to-point/model/switch-mmu.cc

```diff
@@ -345,9 +413,32 @@ bool SwitchMmu::CheckEgressAdmission(uint32_t port, uint32_t qIndex, uint32_t ps
 }
 
 void SwitchMmu::UpdateIngressAdmission(uint32_t port, uint32_t qIndex, uint32_t psize, uint32_t type) {
-	ingress_bytes[port][qIndex] += psize;
-	ingressPoolUsed += psize;
-	totalSharedUsed += psize; // IMPORTANT: totalSharedUsed is only updated in the ingress. No need to update in egress. Avoid double counting.
+	
+	// NOTE: ingress_bytes simple counts total bytes occupied by port, qIndex, 
+	// This includes bytes from ingresspool as well as from headroom. ingress_bytes[port][qIndex] - xoffUsed[port][qIndex] gives us the occupancy in ingressPool.
+	ingress_bytes[port][qIndex] += psize; 
+	totalUsed += psize; // IMPORTANT: totalUsed is only updated in the ingress. No need to update in egress. Avoid double counting.
+
+	/* Update the total headroom used.*/
+	if (type == LOSSLESS){
+		/*First, remove the previously used headroom corresponding to queue: port, qIndex. This will be updated with current value next.*/
+		xoffTotalUsed -= xoffUsed[port][qIndex]; 
+		/*Second, get currently used headroom by the queue: port, qIndex and update `xoffUsed[port][qIndex]` */
+		uint64_t threshold = Threshold(port,qIndex,"ingress",type); // get the threshold
+		/*if headroom is zero*/
+		if (xoffUsed[port][qIndex]==0){ 
+			/*if ingress bytes of the queue exceeds threshold, start using headroom. pfc pause will be triggered by CheckShouldPause later.*/
```

Comment:
It seems that you forget to update ingress pool occupancy.

Reply:
ingress_bytes[port][qIndex] is already updated a few lines above. The total ingress pool occupancy can be accessed by `totalUsed - xoffTotalUsed`